### ISSUE TITLE
openssl fails to connect to some sites on 32-bit

### ISSUE THREAD

**Issue by huglovefan** [ikod/dlang-requests#137]
with this example program:

```D
/+ dub.sdl:
	dependency "requests" version="~>2.0.2"
	debugVersions "requests"
+/
import requests;
import std.experimental.logger;

void main(string[] args)
{
	globalLogLevel = LogLevel.trace;
	getContent(args[1]);
}
```

do a command like `dub run --single --arch=x86 example.d -- URL` where URL is one of:

- `https://www.google.com`: will fail with `requests/streams.d(890): ssl connect failed: certificate verify failed`
- `https://yle.fi`: will fail with `requests/streams.d(890): ssl connect failed: wrong version number`

the same commands will work if `--arch=x86` is removed, or if a different url is used like `https://example.org` (most urls will work, i've only found a few that are affected and don't know what they have in common)

here's the output for google:

```
Running example https://www.google.com
2021-11-21T07:37:41.678 [trace] ssl_adapter.d:132:_sharedStaticCtor_L59_C1 will use libssl.so.1.1
2021-11-21T07:37:41.678 [trace] ssl_adapter.d:139:_sharedStaticCtor_L59_C1 will use libcrypto.so.1.1
2021-11-21T07:37:41.680 [trace] http.d:2400:execute serving Request(GET, https://www.google.com/)
2021-11-21T07:37:41.680 [trace] http.d:2162:exec_from_parameters exec from parameters request
2021-11-21T07:37:41.680 [trace] http.d:2186:exec_from_parameters create new connection
2021-11-21T07:37:41.680 [trace] http.d:625:setupConnection Set up new connection
2021-11-21T07:37:41.680 [trace] http.d:278:select_proxy proxy=null
2021-11-21T07:37:41.680 [trace] streams.d:1097:connect Create connection to www.google.com:443
2021-11-21T07:37:41.700 [trace] streams.d:1106:connect Trying 142.250.74.4:443
object.Exception@../../.dub/packages/requests-2.0.2/requests/source/requests/streams.d(890): ssl connect failed: certificate verify failed
----------------
../../.dub/packages/requests-2.0.2/requests/source/requests/streams.d:890 @trusted void requests.streams.OpenSslSocket.connect(std.socket.Address) [0x5679ca10]
../../.dub/packages/requests-2.0.2/requests/source/requests/streams.d:1115 requests.streams.SocketStream requests.streams.SocketStream.connect(immutable(char)[], ushort, core.time.Duration) [0x5679dc18]
../../.dub/packages/requests-2.0.2/requests/source/requests/streams.d:988 requests.streams.SocketStream requests.streams.SSLSocketStream.connect(immutable(char)[], ushort, core.time.Duration) [0x5679d336]
../../.dub/packages/requests-2.0.2/requests/source/requests/http.d:708 requests.streams.NetworkStream requests.http.HTTPRequest.setupConnection() [0x567b73a4]
../../.dub/packages/requests-2.0.2/requests/source/requests/http.d:2187 requests.http.HTTPResponse requests.http.HTTPRequest.exec_from_parameters() [0x567bcac8]
../../.dub/packages/requests-2.0.2/requests/source/requests/http.d:2409 requests.http.HTTPResponse requests.http.HTTPRequest.execute(requests.request.Request) [0x567be230]
../../.dub/packages/requests-2.0.2/requests/source/requests/request.d:499 requests.base.Response requests.request.Request.LastInterceptor.opCall(requests.request.Request, requests.request.RequestHandler) [0x5678d0c5]
../../.dub/packages/requests-2.0.2/requests/source/requests/request.d:184 requests.base.Response requests.request.RequestHandler.handle(requests.request.Request) [0x5678c1b9]
../../.dub/packages/requests-2.0.2/requests/source/requests/request.d:631 requests.base.Response requests.request.Request.execute(immutable(char)[], immutable(char)[], std.typecons.Tuple!(immutable(char)[], "key", immutable(char)[], "value").Tuple[]) [0x5678df5f]
../../.dub/packages/requests-2.0.2/requests/source/requests/request.d:528 requests.base.Response requests.request.Request.get(immutable(char)[], std.typecons.Tuple!(immutable(char)[], "key", immutable(char)[], "value").Tuple[]) [0x5678d39c]
../../.dub/packages/requests-2.0.2/requests/source/requests/package.d:483 ref requests.streams.Buffer!(ubyte).Buffer requests.getContent(immutable(char)[]) [0x5678bf72]
./app.d:10 _Dmain [0x5678bc34]
Program exited with code 1
```

problem is seen on

- 32-bit (armv7) armbian linux bullseye with ldc 1.24.0 and 1.28.0, openssl 1.1.1k
- 64-bit gentoo linux with `dub --arch=x86`, dmd 2.098.0, openssl 1.1.1l
- 64-bit windows 7 with `dub --arch=x86`, dmd 2.098.0, openssl 1.1.1l (with the example modified to load `cert.pem`)

on the armbian machine, other tools like curl and wget are able to load the sites normally so it shouldn't be a problem related to running 32-bit openssl on a 64-bit system

**Comment by ikod** [comment]
Hello @huglovefan 

Ok, I'l try to set up some kind of 32-bit linux to check. it is possible that I use wrong int type for some call.

Thanks for report

**Comment by ikod** [comment]
@huglovefan 

I reproduced this problem on 32 bit system and even found workaround for https://google.com, but I need some more time to understand how to fix it properly

**Comment by ikod** [comment]
hello @huglovefan 

I commited some fixes, can yoy please check that it works as expected?

**Comment by ikod** [comment]
@huglovefan ping

**Comment by ikod** [comment]
hello, @huglovefan 

can you please confirm that it is fixed(or not)?

**Comment by ikod** [comment]
closed with v2.0.3
